OpenRA.Mods.Common/Widgets/Logic/MusicPlayerLogic.cs

```diff
@@ -36,29 +36,30 @@ public MusicPlayerLogic(Widget widget, Ruleset modRules, World world, Action onE
 			BuildMusicTable();
 
 			Func<bool> noMusic = () => !musicPlaylist.IsMusicAvailable;
+			Func<bool> disableControls = () => !musicPlaylist.IsMusicAvailable || musicPlaylist.AmbientMusic;
```

Comment:
Then this should work, yet if you go into the TD mod's music menu with music installed, you can pretty much pick any song to replace the music.

Thinking of it, I also forgot to override the click-on-songname-to-play-it-action.